recipes/cairomm/all/conanfile.py

```diff
@@ -25,170 +35,118 @@ class CairommConan(ConanFile):
         "fPIC": True,
     }
 
-    generators = "pkg_config"
-    exports_sources = "patches/**"
-    short_paths = True
-
-    def _abi_version(self):
-        return "1.16" if tools.Version(self.version) >= "1.16.0" else "1.0"
-
-    def validate(self):
-        if hasattr(self, "settings_build") and tools.cross_building(self):
-            raise ConanInvalidConfiguration("Cross-building not implemented")
-        if self.settings.compiler.get_safe("cppstd"):
-            if self._abi_version() == "1.16":
-                tools.check_min_cppstd(self, 17)
-            else:
-                tools.check_min_cppstd(self, 11)
-        if self.options.shared and not self.options["cairo"].shared:
-            raise ConanInvalidConfiguration(
-                "Linking against static cairo would cause shared cairomm to link "
-                "against static glib which can cause problems."
-            )
-
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _abi_version(self):
+        return "1.16" if Version(self.version) >= "1.16.0" else "1.0"
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    def _patch_sources(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
-        if is_msvc(self):
-            # when using cpp_std=c++11 the /permissive- flag is added which
-            # attempts enforcing standard conformant c++ code
-            # the problem is that older versions of Windows SDK is not standard
-            # conformant! see:
-            # https://developercommunity.visualstudio.com/t/error-c2760-in-combaseapih-with-windows-sdk-81-and/185399
-            tools.replace_in_file(
-                os.path.join(self._source_subfolder, "meson.build"),
-                "cpp_std=c++", "cpp_std=vc++")
+    def _min_cppstd(self):
+        return 17 if self._abi_version == "1.16" else 11
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
         if self.options.shared:
             self.options["cairo"].shared = True
 
-    def build_requirements(self):
-        self.build_requires("meson/0.59.1")
-        self.build_requires("pkgconf/1.7.4")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("cairo/1.17.4")
-
-        if self._abi_version() == "1.16":
-            self.requires("libsigcpp/3.0.7")
+        self.requires("cairo/1.18.0", transitive_headers=True, transitive_libs=True)
```

Comment:
I see some headers indeed are consuming cairo headers, but is really needed all those transitive_headers and transitive_libs?

Reply:
Yes. The `freetype` library, which is used by the project, is a transitive dependency from `cairo` and is not explicitly declared in `meson.build`, for example.

Reply:
This smells like upstream should then define that as a proper dependency no? Either way, seems like this is the best we can do as things stand, thanks! :)